h5py/_hl/filters.py

```diff
@@ -259,7 +263,23 @@ def rq_tuple(tpl, name):
 
     if chunks is not None:
         plist.set_chunk(chunks)
-        plist.set_fill_time(h5d.FILL_TIME_ALLOC)  # prevent resize glitch
+
+    if fill_time is None:
+        if chunks is None:
+            fill_time = 'ifset'
+        else:
+            fill_time = 'alloc'  # prevent resize glitch
```

Comment:
Are these the same defaults as HDF5 itself? If so, can we skip calling `set_fill_time` entirely in this case, and leave it up to HDF5, so we don't have to ensure we're doing the same thing for all corner cases & with future changes in HDF5.

Reply:
The default from hdf5 is 'ifset', H5D_FILL_TIME_IFSET ([1.10](https://docs.hdfgroup.org/hdf5/v1_10/group___d_c_p_l.html#ga92c5eb5ee19bfd4a9184cf0428d1b00c), [1.12](https://docs.hdfgroup.org/hdf5/v1.12/group___d_c_p_l.html#ga92c5eb5ee19bfd4a9184cf0428d1b00c) and [v1.14](https://docs.hdfgroup.org/hdf5/v1_14/group___d_c_p_l.html#ga92c5eb5ee19bfd4a9184cf0428d1b00c)), and I believe this won't be changed. The original behaviour in h5py is to set H5D_FILL_TIME_ALLOC when it is chunked to 'prevent resize glitch'. I am trying to preserve the original behaviour in h5py (that 'prevent resize glitch' is from 13 years ago!).

Reply:
Ah, sorry, I'd missed this behaviour was already there.

I'm inclined to get rid of that and use HDF5's defaults. There's a good chance whatever it was working round was fixed years ago, and I can't see any further details at a quick look,  so I suspect the only way to find out if it's still needed is to change it and see if anything breaks. We should make sure it's in the release notes if we change the default, though, so it's easy to find the relevant change if something does go wrong.

Reply:
From the comment it looks like the purpose is to prevent access to uninitialised memory. If `max_shape` is `(10,)` and the user just write 5 values, the result of reading the last 5 values later without fill value could be some garbage.  

I can try to remove this behaviour to see if it passes the tests/CI.

Reply:
Tests passed locally if I remove the original h5py behaviour. Let see the CI.